src/components/AnonymousChat/MessageChannelItem.tsx

```diff
@@ -60,8 +71,10 @@ const MessageChannelItem: (props: MessageChannelItemProps) => React.ReactElement
       onPress={onChannelPressed}
       unreadCount={item?.unreadCount}
       isMe={isMe}
-      hasFollowButton={hasFollowButton}
-      handleFollow={handleFollow}
+      hasFollowButton={followStatus?.isFollowing === false || followStatus?.isFollowingFromAction}
+      // hasFollowButton={true}
+      showFollowingButton={followStatus?.isFollowingFromAction}
+      handleFollow={onFollowUser}
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Here, the properties `hasFollowButton` and `showFollowingButton` are being set based on the follow status of the user. The `handleFollow` property is also being set to the `onFollowUser` function. This is a good approach as it allows the follow button to be dynamically displayed based on the follow status, and the follow action can be easily handled by the `onFollowUser` function.
```javascript
hasFollowButton={followStatus?.isFollowing === false || followStatus?.isFollowingFromAction}
showFollowingButton={followStatus?.isFollowingFromAction}
handleFollow={onFollowUser}
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->